src/main/java/com/crowdin/action/DownloadAction.java

```diff
@@ -45,11 +77,31 @@ private void extractTranslations(File archive) {
         }
 
         try {
-            zipFile.extractAll(archive.getParent());
-            NotificationUtil.showInformationMessage("Translations successfully downloaded");
+            zipFile.extractAll(dirPath);
         } catch (ZipException e) {
             NotificationUtil.showInformationMessage("Downloading translations failed");
             e.printStackTrace();
         }
     }
+
+    private List<File> walkDir(Path dir) {
+        try {
+            return java.nio.file.Files.walk(dir)
+                .filter(java.nio.file.Files::isRegularFile)
+                .map(Path::toFile)
+                .collect(Collectors.toList());
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private List<String> filterFiles(List<String> files, List<String> androidCodes) {
+        Pattern pattern = Pattern.compile("[\\\\\\\\\\/]?values-([a-zA-Z-]+)[\\\\\\\\\\/]([^\\\\\\\\|^\\/]+)$");
```

Comment:
Create a static final field. No need to compile pattern each time when this method is invoked

Reply:
Fixed